src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -161,7 +161,13 @@ class HashMap[K, V](initialCapacity: Int, loadFactor: Double)
           else table(indexedHash) = foundNode.next
           contentSize -= 1
 
-        case (None, Some(value)) => put0(key, value, false, hash, indexedHash)
+        case (None, Some(value)) =>
+          val newIndexedHash =
+            if(contentSize + 1 >= threshold) {
+              growTable(table.length * 2)
+              index(hash)
+            } else indexedHash
+          put0(key, value, false, hash, newIndexedHash)
```

Comment:
Should write `getOld = false`, named arg is used inconsistently in this file. This would be an opportunity to amend that. (I was about to contribute a lint and/or scalafix for that, which occurred to me while using rust fsr.)

The previous version could have been:
```
if (contentSize + 1 >= threshold) put0(key, value, hash, getOld = false) else put0(key, value, getOld = false, hash, indexedHash)
```
which also makes it obvious that the parameter order for the `put0` overload is inconsistent. One also notices that `getOld = true` for only one usage; this would be an opportunity to forgo the overload in favor of named variants.

Reply:
Honestly? The code style quality is a bit lacking with this override, there is a ton of inconsistencies around, for example, how nodes are being search for. We have `while` loops alongside `@tailrec` implementing pretty much same scenarios. Also the hole part with `if(contentSize + 1 >= threshold) growTable(table.length * 2)` is being copied all over the place. But the point is that this is actually a critical problem that for some scenarios makes `mutable.HashMap` quite unusable. The way I found this was while debugging poor performance of `mutable.MultiDict` from collection.contrib that utilises `updateWith` quite heavily. 
But I'm new here so not sure what style adjustments I'm allowed to propose.

Reply:
Yes, the only style issue I feel strongly about is having a space between `if` and open paren. /humor. I didn't mean to nit-pick, and you're right that tweaks can wait.

Reply:
Sorry, the missing space was my misprinting.

Reply:
Fixed.